### ISSUE TITLE
ground_truth_matrix error from categorical_accuracy.cpp

### ISSUE THREAD

**Issue by rylberg** [LLNL/lbann#433]
Trying to run Resnet on Pascal, not using Aluminum and running into a ground_truth_matrix error stemming from categorical_accuracy.cpp when lowering mini batch size. "ground_truth matrix has invalid index (row=1000 x col=0)"

**Comment by timmoon10** [comment]
That error happens when the target layer receives all zeros or all NaNs. Either the training is diverging or a layer is misbehaving somewhere. This is the vanilla Resnet in the model zoo?

**Comment by rylberg** [comment]
The problem is occurring on a Resnet I'm working on when I lower the mini-batch size.  I also thought that it may be diverging as well, so I tried lowering the learning rate which resulted in the same error. From what I hear, I am also not the only one receiving this error.

**Comment by ndryden** [comment]
Luke reported a similar thing (see comment in #374 ).

**Comment by timmoon10** [comment]
I suspect this is batch norm behaving badly (see #588). In any case, the error shows up in the deprecated target layer metric (see #507). I'll close this for now and we can reopen if a similar error shows up again.
